<?xml version="1.0" ?>
<robot name="Suchart">

    <material name="black">
        <color rgba="0.0 0.0 0.0 1.0"/>
    </material>

    <material name="orange">
        <color rgba="1.0 0.0 0.0 1.0"/>
    </material>

    <link name="base"/>
    <joint name="fixed" type="fixed">
        <parent link="base"/>
        <child link="link0"/>
    </joint>
    
    <!-- Base Link -->
    <link name="link0">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://suchart_description/stl/AssemBase.STL"/>
            </geometry> 
            <material name="orange"/>
        </visual>
    </link>

    <joint name="joint1" type="revolute">
        <parent link="link0"/>
        <child link="link1"/>
        <origin xyz="0 0 0.0999" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="30" lower="-3.141" upper="3.141" velocity="1.571"/>
    </joint>    

    <!-- Link 1 -->
    <link name="link1">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://suchart_description/stl/AssemBODY.STL"/>
            </geometry> 
            <material name="orange"/>
        </visual>
    </link>

    <joint name="joint2" type="prismatic">
        <parent link="link1"/>
        <child link="link2"/>
        <origin xyz="-0.04412 0 0.10" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="30" lower="-0.10" upper="0.68" velocity="0.02"/>
    </joint>   

    <!-- Link 2 -->
    <link name="link2">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://suchart_description/stl/Joint1.STL"/>
            </geometry> 
            <material name="orange"/>
        </visual>   
    </link>

    <joint name="joint3" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin xyz="0.5 0 0.032" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="30" lower="-2.618" upper="2.618" velocity="1.571"/>
    </joint> 

    <!-- Link 3  -->
    <link name="link3">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://suchart_description/stl/Joint2.STL"/>
            </geometry> 
            <material name="orange"/>
        </visual>   
    </link>

    <joint name="joint4" type="revolute">
        <parent link="link3"/>
        <child link="link4"/>
        <origin xyz="0.32 0 -0.025" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="30" lower="-2.618" upper="2.618" velocity="1.571"/>
    </joint> 

    <!-- Link 4  -->
    <link name="link4">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://suchart_description/stl/joint3.STL"/>
            </geometry> 
            <material name="orange"/>
        </visual>   
    </link>

    <joint name="joint5" type="revolute">
        <parent link="link4"/>
        <child link="link5"/>
        <origin xyz="0 0 -0.025 " rpy="1.571 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="30" lower="-0.3488" upper="1.571" velocity="1.571"/>
    </joint> 

    <link name="link5">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://suchart_description/stl/Link-Gripper.STL"/>
            </geometry> 
            <material name="orange"/>
        </visual>   
    </link>

    <joint name="joint6" type="revolute">
        <parent link="link5"/>
        <child link="gripper"/>
        <origin xyz="0 -0.090 0 " rpy="1.571 0 0"/>
        <axis xyz="0 0 1"/>
        <limit effort="30" lower="-3.14" upper="3.14" velocity="1.571"/>
    </joint> 

    <link name="gripper">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://suchart_description/stl/Gripper-Real.STL"/>
            </geometry> 
            <material name="orange"/>
        </visual>   
    </link>




</robot>